pilot/pkg/xds/filters/filters.go

```diff
@@ -157,6 +151,21 @@ var (
 	HTTPMx = buildHTTPMxFilter()
 )
 
+func BuildRouterFilter(ctx *RouterFilterContext) *hcm.HttpFilter {
```

Comment:
Can you keep the old precompute `Router` when ctx is nil (assuming ctx is nil for cases we don't have overrides?)?

This has a moderate performance impact on allocs:
```
ListenerGeneration/gateways-8                       9.17MB ± 0%        9.55MB ± 0%   +4.13%  (p=0.008 n=5+5)
ListenerGeneration/gateways-shared-8                20.3kB ± 0%        21.1kB ± 0%   +3.70%  (p=0.008 n=5+5)
ListenerGeneration/empty-8                           827kB ± 0%         829kB ± 0%   +0.27%  (p=0.008 n=5+5)
ListenerGeneration/tls-8                             787kB ± 0%         788kB ± 0%   +0.19%  (p=0.008 n=5+5)
ListenerGeneration/telemetry-8                       942kB ± 0%         944kB ± 0%   +0.24%  (p=0.008 n=5+5)
ListenerGeneration/virtualservice-8                 80.7kB ± 0%        81.8kB ± 0%   +1.42%  (p=0.016 n=4+5)
ListenerGeneration/authorizationpolicy-8             623kB ± 0%         624kB ± 0%   +0.18%  (p=0.008 n=5+5)
ListenerGeneration/peerauthentication-8             45.4kB ± 0%        46.1kB ± 0%   +1.66%  (p=0.029 n=4+4)
ListenerGeneration/knative-gateway-8                18.8kB ± 0%        19.6kB ± 0%   +4.06%  (p=0.008 n=5+5)
```